README.md

```diff
@@ -5,64 +5,31 @@ Targets small and large scale private clouds.
 
 # Setting up for development
 
-Generate certificates
+Generate certificates from the head of this repo
 
-    sudo openssl req -nodes -x509 -newkey rsa:2048 -keyout /etc/ssl/private/lxd-ui.key -out /etc/ssl/private/lxd-ui.crt -days 3000
-    sudo cat /etc/ssl/private/lxd-ui.key /etc/ssl/private/lxd-ui.crt | sudo tee -a /etc/ssl/private/lxd-ui.pem
-    sudo cp /etc/ssl/private/lxd-ui.key /etc/ssl/private/lxd-ui.crt.key
-    sudo chown haproxy /etc/ssl/private/lxd-ui.crt.key
-    sudo openssl pkcs12 -keypbe PBE-SHA1-3DES -certpbe PBE-SHA1-3DES -export -in /etc/ssl/private/lxd-ui.crt -inkey /etc/ssl/private/lxd-ui.key -out lxd-ui.pfx -name "LXD UI"
-    sudo chown $USER ~/lxd-ui.pfx
-
-Import the key to you browser
-- chrome linux: browse to chrome://settings/certificates and import lxd-ui.pfx from your user home
-- chrome windows: https://www.comodo.com/support/products/authentication_certs/setup/win_chrome.php#import
-- firefox: http://www.digi-sign.com/support/client%20certificates/import%20firefox
+    cd lxd-ui
+    mkdir -p keys
+    openssl req -nodes -x509 -newkey rsa:2048 -keyout keys/lxd-ui.key -out keys/lxd-ui.crt -days 3000
+    openssl dhparam -out keys/dhparams.pem 2048
+    cat keys/lxd-ui.key keys/lxd-ui.crt > keys/lxd-ui.pem
+    cp keys/lxd-ui.key keys/lxd-ui.crt.key
 
 Setup lxd
 
     snap install lxd
     lxd init # can accept all defaults
     lxc config set core.https_address "[::]:8443"
-    sudo lxc config trust add /etc/ssl/private/lxd-ui.crt
-
-Install HAProxy
-
-    apt install haproxy
-
-Configure HAProxy with below content in /etc/haproxy/haproxy.cfg
-
-    global
-      daemon
-
-    defaults
-      mode  http
-
-    frontend lxd_frontend
-      bind 127.0.0.1:9443 ssl verify required crt /etc/ssl/private/lxd-ui.pem ca-file /etc/ssl/private/lxd-ui.crt
-      acl is_lxd_core path_beg /1.0
-      use_backend lxd_core if is_lxd_core
-      default_backend lxd_ui
-
-    backend lxd_ui
-      server yarn_serve_port 127.0.0.1:3000
-
-    backend lxd_core
-      server lxd_https localhost:8443 ssl verify none crt /etc/ssl/private/lxd-ui.crt
-
-Restart HAProxy
-
-    sudo service haproxy restart
+    sudo lxc config trust add keys/lxd-ui.crt
 
 Install dotrun as described in https://github.com/canonical/dotrun#installation Launch it from the head of this repo
 
     dotrun
 
-Browse through https://localhost:9443/ and **avoid** querying port 3000 directly. Requests to the lxd core won't reach HAProxy on the port 3000.
+You might want to enable a firewall as the last step exposes an api to start or interact with unprivileged containers on your public ip.
```

Comment:
Which step? The last step is launching with `dotrun`, but I'm not sure it's the one you're referring to. Also, since we're here, it might be better to specify that if the selected port is open on your machine (allowed in `ufw` for example) but not open on your router, you expose the API only within the LAN, which is something that might be desired.
In summary, I would rephrase this paragraph so that it clarifies some information (what's the exact step that exposes you to a risk) and includes further useful information about using the API within the LAN and exclude it from the WAN.

Also, I don't understand why this completely replaces the information about how to reach the UI in a browser. `Browse through https://localhost:9443/`should be left there in my opinion.

Reply:
We hide the wrong notification about reaching the ui on port 3000. Also dotrun ends with a message `booting on https://localhost:9443`. So I assumed that is enough.
I just simplified a bit more, so dotrun also generates the keys. Now we end the readme with a not about how to reach the UI again. (see latest commit below)